### PR TITLE
Fix #57910 - Make autofetch period customizable

### LINKED ISSUE #57910: Git: Make autofetch period customizable
Hi

I would love to customize `git.autofetch`'s period in the settings instead of hardcoding three minutes:

https://github.com/Microsoft/vscode/blob/cd7c7c5fa2c16c6e8281436c35e9a7709cfbd89d/extensions/git/src/autofetch.ts#L22

Thanks!

### PR DESCRIPTION
**Pull request by skprabhanjan**
@joaomoreno , here is the PR that fixes #57910 . (@guywald1 take a look at the changes)
Please review it and let me know if anything needs to be changed.
Thanks :)

### CODE REVIEW

File: extensions/git/src/autofetch.ts [hunk 1]
```diff
@@ -113,7 +113,12 @@ export class AutoFetcher {
 				return;
 			}
 
-			const timeout = new Promise(c => setTimeout(c, AutoFetcher.Period));
+			if (Math.max(0, AutoFetcher.Period) === 0) {
```
Comment (joaomoreno) [inline extensions/git/src/autofetch.ts]: We don't need this alert. If it's out of bounds, just set it to 180.

File: src/vs/monaco.d.ts [hunk 2]
```diff
@@ -39,12 +39,17 @@ declare namespace monaco {
 
 
 	export class Promise<T = any> {
-		constructor(executor: (resolve: (value: T | PromiseLike<T>) => void, reject: (reason: any) => void) => void);
+		constructor(
```
Comment (joaomoreno) [inline src/vs/monaco.d.ts]: Please remove these changes.

File: extensions/git/src/autofetch.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ function isRemoteOperation(operation: Operation): boolean {
 
 export class AutoFetcher {
 
-	private static readonly Period = 3 * 60 * 1000 /* three minutes */;
+	private static readonly Period = workspace.getConfiguration('git').get<number>('autoFetchPeriod') as number;
```
Comment (joaomoreno) [inline extensions/git/src/autofetch.ts]: A setting can change while Code is running. You need to listen to the event change.

**Review by joaomoreno — CHANGES_REQUESTED** [review]

File: extensions/git/src/autofetch.ts (hunk 3 above)
Comment (skprabhanjan) [reply extensions/git/src/autofetch.ts]: @joaomoreno , fixed the other two changes and working on this now.
I will close this PR and send a new one when it is completed :) 
Thanks.

**Comment by skprabhanjan** [comment]
Closing this and opening a new PR.

**Pull request closed without merging**